hpat/datatypes/hpat_pandas_series_functions.py

```diff
@@ -1517,18 +1517,18 @@ def hpat_pandas_series_nunique(self, dropna=True):
         raise TypingError('{} The object must be a pandas.series. Given: {}'.format(_func_name, self))
 
     if isinstance(self.data, StringArrayType):
- 
+
         def hpat_pandas_series_nunique_str_impl(self, dropna=True):
             """
             It is better to merge with Numeric branch
-            Unsupported:
-                ['aa', np.nan, 'b', 'b', 'cccc', np.nan, 'ddd', 'dd']
-                [3, np.nan, 'b', 'b', 5.1, np.nan, 'ddd', 'dd']
             """
 
             str_set = set(self._data)
-            return len(str_set)
- 
+            if dropna == False:
+                return len(str_set)
+            else:
+                return len(str_set) + 1
```

Comment:
This should not work with Numba pipeline (in HPAT these tests are disabled).
Input data is String, non None exists and parameter dropna=True
To do such thing I did `nan_exists` variable below

Reply:
@shssf Shouldn't it work at all or is it implemented incorrectly?

Reply:
@Rubtsowa I would expect failed tests